spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/trace/ZipkinProperties.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.trace;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * Zipkin Properties.
+ *
+ * @author Tim Ysewyn
+ * @deprecated Replaced by {@link GcpTraceProperties} since 1.2.
+ * Added for backward & forward compatibility.
+ */
+@Deprecated
+@ConfigurationProperties("spring.zipkin")
+public class ZipkinProperties {
```

Comment:
Why do we need this? It's not part of the original PR.
Also, I don't see where it would be used.
`GcpTraceProperties` already exists in 1.1.

Reply:
Yes, but the property `spring.cloud.gcp.trace.messageTimeout` doesn't exist in 1.1.
Previously `spring.zipkin.messageTimeout` was used to change the timeout.
Since the property and reporter were both part of the zipkin dependency we need to have a way to let the existing configurations work with the new setup since the next service release should be a drop-in replacement.

That's also why I marked it as deprecated and already added the "new" `spring.cloud.gcp.trace.messageTimeout` property to this PR, to be able to move to 1.2.x without any issue in the future.

`spring.cloud.gcp.trace.messageTimeout` takes precedence over `spring.zipkin.messageTimeout` if you configure both.

Reply:
It's used here: https://github.com/spring-cloud/spring-cloud-gcp/pull/1744/files#diff-fc1b9a6100070e7eb2d92338e77686e5R140

Reply:
I think I understand what you're saying. However, I would avoid introducing a properties class with a namespace that we do not own (`spring.zipkin`). You can instead auto-wire the `Environment` and read the value of `spring.zipkin.messageTimeout` directly from the environment.